specification/src/chapter5.adoc

```diff
@@ -2,6 +2,66 @@
 
 == Cryptography
 
-<use models enabled by crypto ISA extensions>
+RISC-V supports a number of ISA-level extensions aimed at improving performance and reducing the risk of side-channel attacks for cryptographic operations (scalar and vector). They also include an ISA-level entropy source, and guidelines for data independent execution latency.
 
-<crypto guidelines>
\ No newline at end of file
+See xref:chapter3.adoc#_cryptography[cryptography]
+See https://github.com/riscv/riscv-crypto
+
+Work is on-going to extend ISA-level crypto extensions to support operations used in _post-quantum cryptography (PQ)_, and to add key management support - _high assurance cryptography (HAC)_.
+
+Cryptographic requirements depend on target ecosystem, as well as on varying regulatory requirements in different geographic regions. This chapter summarizes commonly used cryptographic guidance for secure systems, provided as guidance for development of RISC-V specifications and RISC-V based secure systems.
+
+=== PQC readiness
+
+Quantum safe cryptography is an evolving area of research. Strong candidates exist with different trade-offs for efficiency, performance, key sizes and other properties. For example, see:  https://csrc.nist.gov/projects/post-quantum-cryptography.
+
+RISC-V systems and specifications must at least support a migration path towards use of PQC. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#
+| Requirement
+
+| CAT_NNN
+| All immutable components MUST use quantum safe cryptography
+
+| CAT_NNN
+| All mutable components MUST at least have a migration path to quantum safe cryptography.
+
+|===
+
+For example, immutable boot code must use a quantum safe algorithm for verification of a mutable boot stage, as an immutable boot stage cannot be updated.
+
+Mutable stages can be updated, and can provide a migration path to quantum safe cryptography. For example, system designers should consider protocols, governance, and storage requirements for upgrading hardware provisioned assets to PQC versions.
+
+=== Cryptographic algorithms and guidelines
+
+The following resources provide general cryptographic guidance applicable to most western jurisdictions:
+https://csrc.nist.gov/Projects/Cryptographic-Standards-and-Guidelines
+https://www.cnss.gov/CNSS/issuances/Memoranda.cfm
+
+In particular, for most new systems:
+
+* Public identifier: 512 bits (for example, hash of a public key)
+* Counter used as identifier: 64 bits
+* Block cipher: AES-256
+* Hash function: SHA-512, or SHA-3
+* Message authentication: HMAC-SHA-512
+* Asymmetric: RSA-3072, or ECC-384
+
+Some legacy use cases may require use of other algorithms, such as SHA-256 or AES-128. In these cases, wherever possible, an upgrade path should be supported. For example, allocating sufficient storage to accommodate larger sizes in future updates.
+
+Some use cases, such as cryptographic memory protection, may sometimes use specialized algorithms not discussed here.
```

Comment:
why not specify some standards such as AES-XTS or at least key strengths?

Reply:
I don't think this document is the place to specify details about cryptographic memory protection. It is a system function and there are many options depending on ecosystem requirements.

I rephrased the sentence.